### PR TITLE
PUBDEV-8634: Fix R check length == 1 for logic operations

### PR DESCRIPTION
**Pull request by tomasfryda**
https://h2oai.atlassian.net/browse/PUBDEV-8634

### CODE REVIEW

File: h2o-r/h2o-package/R/kvstore.R [hunk 1]
```diff
@@ -257,11 +257,10 @@ h2o.getModel <- function(model_id) {
...
       if(type == "numeric" && class(value) == "list" && length(value) == 0) #Special case when using deep learning with 0 hidden units
         value <- 0
-      else if (type == "numeric" && value == "Infinity")
-        value <- Inf
-      else if (type == "numeric" && value == "-Infinity")
-        value <- -Inf
-
+      else if (type == "numeric") {
+        value[value == "Infinity"] <- Inf
+        value[value == "-Infinity"] <- -Inf
```
Comment (tomasfryda) [inline h2o-r/h2o-package/R/kvstore.R:262]: Sadly, if the "±Infinity" is present the vector is a character vector. First I tried to convert it to numeric but it failed on some test due to the fact that the number was so big that it wouldn't fit in R's integer and was expected to be a character vector.

I haven't come up with a better solution than this one yet.

**Review by michalkurka — APPROVED** [review]
LGTM - thank you @tomasfryda

File: h2o-r/h2o-package/R/kvstore.R (hunk 1 above)
Comment (tomasfryda) [reply h2o-r/h2o-package/R/kvstore.R:262]: So far, I have come up only with this not so pretty solution:
```r
        # If there is no loss of information for integers, convert to numeric
        num_value <- as.numeric(value)
        if (is.character(value) && all(
          grepl(".", value, fixed = TRUE) | # Not an integer
          !is.finite(num_value) | # Not a finite number (NaN/Inf)
          (-2^.Machine$double.digits <= num_value & # Lies between the min and max fully representable number
            num_value >= 2^.Machine$double.digits)))
          value <- num_value  
```

Comment (ledell) [reply h2o-r/h2o-package/R/kvstore.R:262]: If it works, it's good for me!  I don't mind if it's a bit complicated; it gets the job done. 👍

**Review by ledell — APPROVED** [review]
I tested it and it seems to fix the error, thank you @tomasfryda!

**Pull request merged**